src/lib/pickles/wrap.ml

```diff
@@ -283,518 +283,391 @@ let deferred_values (type n) ~(sgs : (Backend.Tick.Curve.Affine.t, n) Vector.t)
-             from two proofs back! (e.g. the deferred values)
-           * The next step proof also computes the deferred values inside the circuit and verifies
-             that they match those used by the previous wrap proof.
-
-      The code below generates the deferred values so that we can verifiy that we can actually
-      compute those values correctly inside the circuit.  Special thanks to Matthew Ryan for
-      explaining this in detail. *)
-  let { deferred_values; x_hat_evals; sponge_digest_before_evaluations } =
-    deferred_values ~feature_flags ~actual_feature_flags ~sgs:[]
-      ~prev_challenges:[] ~step_vk:vk ~public_input:[] ~proof
-      ~actual_proofs_verified:Nat.N0.n
-  in
-
-  (* Define Typ.t for Deferred_values.t -- A Type.t defines how to convert a value of some type
-                                          in OCaml into a var in circuit/Snarky.
-
-     This complex function is called with two sets of inputs: once for the step circuit and
-     once for the wrap circuit.  It was decided not to use a functor for this. *)
-  let deferred_values_typ =
-    let open Impls.Step in
-    let open Step_main_inputs in
-    let open Step_verifier in
-    Wrap.Proof_state.Deferred_values.In_circuit.typ
-      (module Impls.Step)
-      ~feature_flags ~challenge:Challenge.typ ~scalar_challenge:Challenge.typ
-      ~dummy_scalar:(Shifted_value.Type1.Shifted_value Field.Constant.zero)
-      ~dummy_scalar_challenge:
-        (Kimchi_backend_common.Scalar_challenge.create
-           Limb_vector.Challenge.Constant.zero )
-      (Shifted_value.Type1.typ Field.typ)
-      (Branch_data.typ
-         (module Impl)
-         ~assert_16_bits:(Step_verifier.assert_n_bits ~n:16) )
-  in
-
-  (* Use deferred_values_typ and the constant helper to prepare deferred_values
-     for use in the circuit.  We change some Opt.t to Option.t because that is
-     what Type.t is configured to accept. *)
-  let deferred_values =
-    constant deferred_values_typ
-      { deferred_values with
-        plonk =
-          { deferred_values.plonk with
-            lookup = Opt.to_option deferred_values.plonk.lookup
-          ; optional_column_scalars =
-              Composition_types.Wrap.Proof_state.Deferred_values.Plonk
-              .In_circuit
-              .Optional_column_scalars
-              .map ~f:Opt.to_option
-                deferred_values.plonk.optional_column_scalars
-          }
-      }
-  (* Prepare all of the evaluations (i.e. all of the columns in the proof that we open)
-     for use in the circuit *)
-  and evals =
-    constant
-      (Plonk_types.All_evals.typ (module Impls.Step) feature_flags)
-      { evals = { public_input = x_hat_evals; evals = proof.openings.evals }
-      ; ft_eval1 = proof.openings.ft_eval1
+let%test_module "gate finalization" =
+  ( module struct
+    type feature_flags = Plonk_types.Opt.Flag.t Plonk_types.Features.t
```

Comment:
this is a tuple type? (asking basic ocaml stuff to understand the syntax better)

Reply:
Ooooh! thank you makes sense now